### DISCUSSION TITLE
Possible to control strip/cell spacing?

### DISCUSSION THREAD

**Discussion by andersforsgren** [emilk/egui#2186] [Q&A]
Is it possible in any way to control the spacing between strips/cells for a StripBuilder?

e.g here is a two column layout where the first column has two rows. 
As an example, the desired spacing is 12 between the columns, 0 between the rows in the column, and 8 between the items inside the cell. 

The outermost and innermost spacings are possible to set, (can set before starting the stripbuilder, and inside the cell).  But not "in the middle of it". Is this a misunderstanding on my part? 
I can add empty "spacing rows/columns", but it seems like there should be an easier way. 

```rust
  ui.spacing_mut().item_spacing = egui::vec2(12.0, 12.0);   // OK: can set a spacing before creating the builder

  StripBuilder::new(ui)
      .size(Size::abolute(100.0))
      .size(Size::remainder())
      .vertical(|mut strip| {                               // Spacing 12 used to separate the two columns       
          strip.strip(|mut builder| {
              builder
                 .size(Size::relative(0.5))
                 .size(Size::relative(0.5))
                 .horizontal(|mut strip| {                  // Want zero spacing between strip rows here
                      strip.ui.spacing_mut().item_spacing = // ... but ui is inaccessible
                      strip.cell(|ui| {
                                                            // ...until inside a cell, where ui is once again exposed
                           ui.spacing_mut().item_spacing = egui::vec2(8.0, 8.0);  
                           stuff(ui);
                      });
                      strip.cell(|ui| {
                           ui.spacing_mut().item_spacing = egui::vec2(8.0, 8.0);  // 
                           more_stuff(ui);
                      });
                      ...
```

**Comment by ls-ivan-kiselev** [comment]
It's generally problematic for me to not have UI exposed on `Strip` level as well, as I want to have granular control of spacing within one strip, thus I have multiple cells, but I want the entire strip to behave like a single object for the user, e.g. selection/hover animation over any member of a strip. 

Think of list UIs like this: 
![image](https://user-images.githubusercontent.com/86964964/209946614-e6197879-3da2-4b80-ad0b-240c7329086c.png)


It's important that all the elements within every list item are spaced the same, so I am using `Stripes` for them, but overall they should behave as a single item for the user, thus it would be convenient to have `Ui` exposed on the stripe level as much as on the cell level

**Comment by juanespj** [comment]
I found using Grid layouts way simpler to achieve what you want, if you want different spacings you can control it by using multiple Grid instances or ui.horizontal / ui. vertical to group items.

```rust
egui::Grid::new("detrep_month_det")
                    .spacing(egui::Vec2::new(10.0, 2.0))
                    .striped(true)
                    .show(ui, |ui| {
ui.label("row1");
ui.label("row1");
ui.label("row1");
ui.endrow();
ui.label("row1");
ui.label("row1");
ui.label("row1");
///ui here
});
```
